consoles/VMWare.pm

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package consoles::VMWare;
+
+use Mojo::Base -base, -signatures;
+
+use Mojo::JSON qw(encode_json);
+use Mojo::UserAgent;
+use Mojo::URL;
+use Mojo::Util qw(xml_escape);
+
+use bmwqemu;
+use log;
+
+has protocol => 'https';
+has host => undef;
+has vm_id => undef;
+has username => 'root';
+has password => undef;
+has dewebsockify_pid => undef;
+
+sub _get_vmware_error ($dom) {
+    my $faultstring_element = $dom->find('faultstring')->first;
+    return $faultstring_element ? $faultstring_element->text : '';
+}
+
+sub _prepare_vmware_request ($ua, $api_url, $xml) {
+    my $txn = $ua->build_tx(POST => $api_url);
+    my $headers = $txn->req->headers;
+    $txn->req->body($xml);
+    $headers->header(SOAPAction => 'urn:vim25/7.0.2.0');
+    $headers->content_type('text/xml');
+    $headers->content_length(length $xml);
+    return ($txn, $headers);
+}
+
+sub configure_from_url ($self, $url) {
+    $url = Mojo::URL->new($url);
+    $self->protocol($url->protocol)->host($url->host);
+    $self->username($url->username)->password($url->password);
+    $self->vm_id(substr($url->path, 1)) if length $url->path > 1;
+}
+
+sub get_vmware_wss_url ($self) {
+
+    # make XML for requests
+    my $protocol = $self->protocol or die "No protocol specified\n";
+    my $host = $self->host or die "No VMWare host specified\n";
+    my $api_url = "$protocol://$host/sdk";
+    my $username = xml_escape $self->username;
+    my $password = xml_escape $self->password;
+    my $vm_id = xml_escape($self->vm_id || $bmwqemu::vars{VIRSH_VM_ID});
+    my $auth_xml = qq{<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><Header><operationID>esxui-8fb8</operationID></Header><Body><Login xmlns="urn:vim25"><_this type="SessionManager">ha-sessionmgr</_this><userName>$username</userName><password>$password</password></Login></Body></Envelope>};
+    my $request_wss_xml = qq{<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><Header><operationID>esxui-c51d</operationID></Header><Body><AcquireTicket xmlns="urn:vim25"><_this type="VirtualMachine">$vm_id</_this><ticketType>webmks</ticketType></AcquireTicket></Body></Envelope>};
+
+    # request VMWare session
+    my $ua = ($self->{_vmware_ua} //= Mojo::UserAgent->new);
+    $ua->cookie_jar->empty;    # avoid auth error because we're already logged in
+    my ($auth_txn, $auth_headers) = _prepare_vmware_request($ua, $api_url, $auth_xml);
+    $auth_headers->cookie('vmware_client=VMware');
+    $ua->insecure(1);    # so far our setup doesn't have a proper certificate
```

Comment:
This might seem like nit-picking but I would prefer if this was explicitly requested rather than done automatically. If I was setting up a machine I would expect to see an error if the certificate wasn't valid